metadata/src/main/java/com/redhat/lightblue/metadata/Index.java

```diff
@@ -135,4 +137,13 @@ public boolean isCaseInsensitiveKey(Path path) {
                 .filter(IndexSortKey::isCaseInsensitive)
                 .anyMatch(i -> i.getField().equals(path));
     }
+
+    /**
+     *
+     * @return arbitrary index options, can be backend specific
+     */
+    public Properties getOptions() {
```

Comment:
You don't really need this. Index extends MetadataObject, and it contains a Properties property. Any unrecognized property will be placed into that properties object as an object map.